### PR TITLE
Fixes vendor showing weird glass icons

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Fixes vendor showing weird glass icons
* original sorce: https://github.com/tgstation/tgstation/pull/71929
* closes https://github.com/Skyrat-SS13/Skyrat-tg/pull/18242

since skyrat has its own addition, the fix had to be made with an additional touch.
## Why It's Good For The Game
bug fix good
## Changelog
:cl:
fix: vendor no longer shows glass icons in a specific condition
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
image

**Comment by EvilDragonfiend** [comment]
I removed a wrong branch lol

File: code/modules/asset_cache/assets/vending.dm [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+/datum/asset/spritesheet/vending/create_spritesheets()	// initialising the list of items we need
+	var/target_items = list()
+	for(var/obj/machinery/vending/vendor as anything in typesof(/obj/machinery/vending))
+		vendor = new vendor() // It seems `initial(list var)` has nothing. need to make a type.
+		for(var/each in list(vendor.products, vendor.premium, vendor.contraband))
+			target_items |= each
+		qdel(vendor)
+
+	// building icons for each item
+	for (var/k in target_items)
```
Comment (softcerv) [inline code/modules/asset_cache/assets/vending.dm:13]: If you are going to make non-modular changes, please make sure to have the original code saved as a comment

File: code/modules/vending/_vending.dm [hunk 2]
```diff
@@ -329,7 +329,6 @@
 			if (dump_amount >= 16)
 				return
 
-GLOBAL_LIST_EMPTY(vending_products)
```
Comment (softcerv) [inline code/modules/vending/_vending.dm:332]: again, please save this as a comment

File: code/modules/vending/_vending.dm [hunk 3]
```diff
@@ -357,7 +356,6 @@ GLOBAL_LIST_EMPTY(vending_products)
 
 		var/obj/item/temp = typepath
 		var/datum/data/vending_product/R = new /datum/data/vending_product()
-		GLOB.vending_products[typepath] = 1
```
Comment (softcerv) [inline code/modules/vending/_vending.dm:360]: again, please save this as a comment

**Comment by Funce** [comment]
I'm going to merge the changes here into the mirror PR

**Comment by Funce** [comment]
Thank you for the help with this! <3

**Pull request closed without merging**